apps/ui-builder/src/components/ComponentTreeItem.tsx

```diff
@@ -0,0 +1,220 @@
+    if (selected) {
+      return;
+    }
+    handleSelectComponent(node.element);
+  }, [handleSelectComponent, node, selected]);
+
+  const onKeyDown = React.useCallback(
+    e => {
+      // enter or space
+      if (e.key === 'Enter' || e.keyCode === 32) {
+        selectItem();
+        return;
+      }
+
+      if (e.key === 'ArrowDown') {
+        e.target.nextSibling && e.target.nextSibling.focus();
+        return;
+      }
+      if (e.key === 'ArrowUp') {
+        e.target.previousSibling && e.target.previousSibling.focus();
+        return;
+      }
+
+      if (isMac) {
+        macKeyDown(e, e.target);
+      }
+    },
+    [selectItem],
+  );
+
+  React.useEffect(() => {
+    if (!selected) {
+      setDeletePopoverOpened(false);
+    }
+  }, [selected]);
+
+  const deleteButton = React.useMemo(
+    () => (
+      <Popover open={deletePopoverOpened}>
+        <PopoverTrigger>
+          <Tooltip relationship="label" content="Delete" withArrow>
+            <Button
+              icon={<DeleteRegular />}
+              appearance="subtle"
+              size="small"
+              onClick={openPopover}
+              className={styles.button}
+            />
+          </Tooltip>
+        </PopoverTrigger>
+        <PopoverSurface className={styles.deletePopover}>
+          <span className={styles.deletePopoverCaption}>Are you sure you want to delete {node.title}?</span>
+          <Button onClick={closePopover}>Cancel</Button>
+          <Button appearance="primary" onClick={handleDeleteSelected}>
+            Delete
+          </Button>
+        </PopoverSurface>
+      </Popover>
+    ),
+    [node, deletePopoverOpened, openPopover, closePopover, handleDeleteSelected, styles],
+  );
+
+  const onMouseEnter = React.useCallback(() => setOnHover(true), [setOnHover]);
+  const onMouseLeave = React.useCallback(() => setOnHover(true), [setOnHover]);
+
+  const addAfter = React.useCallback(() => handleAddComponent(node.id, 'after'), [handleAddComponent, node.id]);
+  const addBefore = React.useCallback(() => handleAddComponent(node.id, 'before'), [handleAddComponent, node.id]);
+  const addChild = React.useCallback(() => handleAddComponent(node.id, 'child'), [handleAddComponent, node.id]);
+  const remove = React.useCallback(() => handleDeleteComponent(node.id), [handleDeleteComponent, node.id]);
+
+  return (
+    <Menu openOnContext={true}>
+      <MenuTrigger>
+        <div
+          tabIndex={0}
+          onKeyDown={onKeyDown}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+          onClick={selectItem}
+          style={{ paddingLeft: `${node.level * 0.5}rem` }}
+          className={mergeClasses(styles.treeItem, selected && styles.selected)}
+        >
+          <span className={styles.title}>{node.title}</span>
+          {(selected || onHover) && (
+            <div>
+              <Tooltip relationship="label" content="Move" withArrow>
+                <Button
```

Comment:
you are putting one focusable element into another, this can be confusing for screen reader users. before, the actions were not focusable and were made available using context menu.

Reply:
Context menu items are not exactly the same as buttons here; I will remove focusability from here and add copy and move options into menu